server/src/main/java/io/crate/execution/engine/aggregation/impl/CmpByAggregation.java

```diff
@@ -171,4 +250,82 @@ public Object terminatePartial(RamAccounting ramAccounting, CompareBy state) {
     public DataType<?> partialType() {
         return partialType;
     }
+
+
+    static class CmpByLongState {
+
+        static final long SHALLOW_SIZE = RamUsageEstimator.shallowSizeOfInstance(CmpByLongState.class);
+
+        long cmpValue = Long.MIN_VALUE;
+
+        int docId;
+        LeafReaderContext leafReaderContext;
+        boolean hasValue = false;
+    }
+
+    static class CmpByLong implements DocValueAggregator<CmpByLongState> {
+
+        private final int mod;
+        private final String columnName;
+        private final LuceneCollectorExpression<?> resultExpression;
+        private final DataType<?> searchType;
+
+        private SortedNumericDocValues values;
+        private LeafReaderContext leafReaderContext;
+
+        public CmpByLong(int mod,
+                         String columnName,
+                         DataType<?> searchType,
+                         LuceneCollectorExpression<?> resultExpression) {
+            this.mod = mod;
+            this.columnName = columnName;
+            this.searchType = searchType;
+            this.resultExpression = resultExpression;
+            resultExpression.startCollect(new CollectorContext());
+        }
+
+        @Override
+        public CmpByLongState initialState(RamAccounting ramAccounting,
+                                           MemoryManager memoryManager,
+                                           Version minNodeVersion) {
+            ramAccounting.addBytes(CmpByLongState.SHALLOW_SIZE);
+            return new CmpByLongState();
+        }
+
+        @Override
+        public void loadDocValues(LeafReaderContext leafReaderContext) throws IOException {
+            this.leafReaderContext = leafReaderContext;
+            values = DocValues.getSortedNumeric(leafReaderContext.reader(), columnName);
+        }
+
+        @Override
+        public void apply(RamAccounting ramAccounting, int doc, CmpByLongState state) throws IOException {
+            if (values.advanceExact(doc) && values.docValueCount() == 1) {
```

Comment:
`values.docValueCount() == 1` means not an array, right?
What happens if there is an array, do we guard this in analysis already?